src/Parser/Shaman/Elemental/Modules/Talents/ElementalBlast.js

```diff
@@ -14,6 +14,26 @@ class ElementalBlast extends Analyzer {
     combatants: Combatants,
   };
 
+  currentBuffAmount=0;
+  lastFreshApply=0;
+  resultDuration=0;
+
+  on_removebuff(event) {
+    if (event.ability.guid === SPELLS.ELEMENTAL_BLAST_CRIT.id ||event.ability.guid === SPELLS.ELEMENTAL_BLAST_MASTERY.id ||event.ability.guid === SPELLS.ELEMENTAL_BLAST_HASTE.id){
+      this.currentBuffAmount--;
+      if (this.currentBuffAmount===0)
+        this.resultDuration+=event.timestamp-this.lastFreshApply;
+    }
+  }
+
+  on_applybuff(event) {
+    if (event.ability.guid === SPELLS.ELEMENTAL_BLAST_CRIT.id ||event.ability.guid === SPELLS.ELEMENTAL_BLAST_MASTERY.id ||event.ability.guid === SPELLS.ELEMENTAL_BLAST_HASTE.id){
+      if (this.currentBuffAmount===0)
+        this.lastFreshApply=event.timestamp;
+      this.currentBuffAmount++;
+    }
+  }
+
   on_initialized() {
```

Comment:
I would prefer having that as a first method in the class for readability but it's just a suggestion, not an issue.